server/src/api/roadmaps/roadmaps.controller.ts

```diff
@@ -96,3 +96,34 @@ export const deleteRoadmaps: RouteHandlerFnc = async (ctx) => {
 
   ctx.status = numDeleted == 1 ? 200 : 404;
 };
+
+export const leaveRoadmap: RouteHandlerFnc = async (ctx) => {
+  if (!ctx.state.user) throw new Error('User is required');
+  await Role.transaction(async (trx) => {
+    const userRole = await Role.query(trx)
+      .where('userId', ctx.state.user!.id)
+      .andWhere('roadmapId', ctx.params.roadmapId)
+      .first();
```

Comment:
```suggestion
      .findById([ctx.state.user!.id, Number(ctx.params.roadmapId)]);
```

Should also check if the role was found